mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -616,6 +616,50 @@ void CMomentumPlayer::Teleport(const Vector *newPosition, const QAngle *newAngle
     // No need to remove the trail here, CreateTrail() already does it for us
     BaseClass::Teleport(newPosition, newAngles, newVelocity);
     CreateTrail();
+
+    g_ReplaySystem.SetTeleportedThisFrame();
+}
+
+bool CMomentumPlayer::KeyValue(const char *szKeyName, const char *szValue)
+{
+    //
+    // It's possible for some maps to use "AddOutput -> origin x y z
+    // for teleports.
+    //
+    if (FStrEq(szKeyName, "origin"))
+    {
+        // Copy of CBaseEntity::KeyValue
+        Vector vecOrigin;
+        UTIL_StringToVector(vecOrigin.Base(), szValue);
+
+        // If you're hitting this assert, it's probably because you're
+        // calling SetLocalOrigin from within a KeyValues method.. use SetAbsOrigin instead!
+        Assert((GetMoveParent() == NULL) && !IsEFlagSet(EFL_DIRTY_ABSTRANSFORM));
+        SetAbsOrigin(vecOrigin);
+
+
+
+        // TODO: Throw this into OnTeleport() or something?
+        CreateTrail();
+
+        g_ReplaySystem.SetTeleportedThisFrame();
+
+        return true;
+    }
+
+    return BaseClass::KeyValue(szKeyName, szValue);
+}
+
+// These two are overridden because of a weird compiler error,
```

Comment:
Interesting, I'll trust that because I've seen weird shit like this too before, but I was able to implement overriding just the string version just fine for triggers in zone-menu branch. What error were you getting?

Otherwise it seems ready for merge.

Reply:
[It's this warning](https://www.dropbox.com/s/tnovxdtefhlqyly/momwarning.png?dl=0) when you remove those overrides.
Never seen that error before and makes no sense to me.

Reply:
Meh alright, we'll keep it.